core/sched.cc

```diff
@@ -350,16 +372,18 @@ void cpu::reschedule_from_interrupt(bool called_from_yield,
         n->_runtime.add_context_switch_penalty();
     }
     preemption_timer.cancel();
-    if (!called_from_yield) {
-        if (!runqueue.empty()) {
-            auto& t = *runqueue.begin();
-            auto delta = n->_runtime.time_until(t._runtime.get_local());
-            if (delta > 0) {
-                preemption_timer.set_with_irq_disabled(now + delta);
+    if (n->_realtime._priority == 0) {
```

Comment:
It looks like I am completely ignorant here, but why execute the same logic - the `if (!called_from_yield) {` block and it's else` body ONLY if the priority of the next thread scheduled `n` is equal to 0? Is it because by design (POSIX) the runtime does not matter for real-time threads (aka priority > 0)?

Reply:
The code below deals with the preemption timer. Currently (only SCHED_FIFO implemented, not SCHED_RR) only non-realtime (priority==0) have preemption timer. Such threads need to calculate how much time will pass until the "runtime" of this thread will reach the runtime of the next thread - and set the preemption timer to that time. None of this is needed for real-time threads which just run in FIFO order, with no preemptions and no concept of "runtime" (how much time this thread recently ran).